### DISCUSSION TITLE
Trouble with accessing to the model of Route Model Binding

### DISCUSSION THREAD

**Discussion by StambulaDmitriy** [livewire/livewire#3165] [1. Help]
I have a component on a page with Route Model Binding. It isn`t a full page component. It contains only one button for refreshing. 

**routes/web.php:**
```
Route::get('categories/{category}', function(Category $category) {
    return view('categories.view', compact($category));
});
```

**views/categories/view.blade.php:**

```
@extends('layouts.main')

@section('content')

    <h1>{{ $category->name }}</h1>
    <p>{{ $category->description }}</p>

    @livewire('category-products')

@endsection
```

**app/http/livewire/CategoryProducts.php:**

```
class CategoryProducts extends Component
{
    public function render()
    {
        return <<<'blade'
            <div>
                <button wire:click="$refresh">Refresh</button>
            </div>
        blade;
    }
}
```

Every time component is re-rendering (refreshing) I see in the debug bar of this AJAX request (https://github.com/barryvdh/laravel-debugbar) a query:

![Снимок экрана от 2021-06-22 10-58-28](https://user-images.githubusercontent.com/52291312/122886840-fcf5b800-d348-11eb-9f74-a2e05a768a7a.png)

**How do I access to the result of this query in the component every time the component is re-rendering?**

**Comment by joshhanley** [comment]
@StambulaDmitriy you need to pass the category into the component, like 
```blade
@livewire('category-products', ['category' => $category)
```

Then in your `category-products` class, add a public property called category, which Livewire will populate for you from the above call
```php
public $category;
```

See here for details https://laravel-livewire.com/docs/2.x/rendering-components#:~:text=%3E-,parameters,-Passing

Hope this helps!

**Reply by StambulaDmitriy** [reply]
Hi! 
Thank you for your answer! But this doesn't fit in my case. I need access to a model object to access relations.
In the solution that you suggested, the model object I will receive only when the component is initialized, in the future I will have an associative array instead of a model. The only option for me is to pass the category id at the initialization of the component and perform a query in the database each time it is rendered. But! Such a request is already being done under the hood of Livewire (it is shown in the screenshot of debugbar) and I am wondering how to access the results of this request.

**Reply by joshhanley** [reply]
@StambulaDmitriy I suggest trying it, any Eloquent models assigned to public properties will be re-hydrated to a model instance, not an array so you can use relations and properties as you need 🙂

But personally, I wouldn't actually use a Laravel controller in that instance, Livewire supports route model binding using full page components, so that is how I would set it up. See here https://laravel-livewire.com/docs/2.x/rendering-components#route-model-binding

Hope this helps!

**Reply by StambulaDmitriy** [reply]
I am unable to use the full page component due to problems with the breadcrumb generator. How can I get rid of this unexpected database query?

**Reply by joshhanley** [reply]
@StambulaDmitriy ah right, hmm I don't think you can.

My understanding is, due to some recent security changes, Livewire now polls the route, that the LW component came from originally (even if it was nested inside a standard Laravel controller route), to check for middleware and a few other things. 

I believe that Laravel's implicit route model binding is making that query when LW inspects the route, hence that query. So I'm not sure if you can get rid of that. Is that one query really an issue?
